pinot-core/src/main/java/com/linkedin/pinot/core/startree/OffHeapStarTreeBuilder.java

```diff
@@ -269,26 +284,32 @@ public void append(GenericRow row) throws Exception {
       }
       dimension.setDimension(i, dictionary.get(dimValue));
     }
+    // initialize raw data row
     Number[] numbers = new Number[numMetrics];
+    byte[] hllColumnInBytes = null;
     for (int i = 0; i < numMetrics; i++) {
       String metName = metricNames.get(i);
-      numbers[i] = (Number) row.getValue(metName);
+      if (i == hllColumnIndex) { // TODO: check if it actually supports byte[] type
+        hllColumnInBytes = (byte[]) row.getValue(metName);
+      } else {
+        numbers[i] = (Number) row.getValue(metName);
+      }
     }
-    MetricBuffer metrics = new MetricBuffer(numbers);
-    append(dimension, metrics);
-  }
-
-  public void append(DimensionBuffer dimension, MetricBuffer metrics) throws Exception {
-    appendToRawBuffer(dimension, metrics);
+    MetricBuffer metricBuffer;
+    if (hllColumnInBytes != null) {
+      metricBuffer = new MetricBuffer(numbers, hllColumnIndex, hllColumnInBytes);
```

Comment:
instead of if-else look at adding hll as a DerivedColumn that takes in raw column as input and generates a derived column